src/main/java/ca/on/oicr/gsi/dimsum/service/CaseService.java

```diff
@@ -557,6 +504,22 @@ private Stream<TestTableView> filterTestTableViews(List<Case> cases,
     return stream;
   }
 
+  private Map<Case, List<Test>> getFilteredCaseAndTest(List<Case> cases,
+      Collection<CaseFilter> filters) {
+    Map<Case, List<Test>> testsByCase = new HashMap<>();
+    List<Case> filteredCases = filterCases(cases, filters).toList();
+    if (filters != null && !filters.isEmpty()) {
+      Map<CaseFilterKey, Predicate<Test>> filterMap =
+          buildFilterMap(filters, CaseFilter::testPredicate);
+      for (Predicate<Test> predicate : filterMap.values()) {
+        for (Case kase : filteredCases) {
+          testsByCase.put(kase, kase.getTests().stream().filter(predicate).toList());
+        }
+      }
+    }
+    return testsByCase;
```

Comment:
This would return an empty map if called with no filters. Change the method to validate `filters` at the start since this should never happen. Throw NullPointerException if null, or IllegalArgumentException if empty